ext/pdo/pdo_stmt.c

```diff
@@ -688,13 +634,84 @@ static bool pdo_do_key_pair_fetch(pdo_stmt_t *stmt, enum pdo_fetch_orientation o
 	return true;
 }
 
+/* Previous the legacy zend_fcall_info_args_ex() function was used to set the
+ * constructor arguments. The particularity of this function is to it "wraps" arguments
+ * that need to be passed by-ref in a reference.
+ * As this API only supported positional args, we don't bother doing this for named arguments. */
+static HashTable* pdo_class_fetch_prepare_constructor_arguments(const zend_function *constructor, HashTable *ctor_args)
+{
+	zend_ulong index;
+	zend_string *arg_name = NULL;
+	zval *arg_value = NULL;
+	bool needs_duplication = false;
+	ZEND_HASH_FOREACH_KEY_VAL(ctor_args, index, arg_name, arg_value) {
+		/* Named param, means we can assume no positional arguments exist after */
+		if (arg_name) {
+			break;
+		}
+		if (!Z_ISREF_P(arg_value) && ARG_SHOULD_BE_SENT_BY_REF(constructor, index+1)) {
+			needs_duplication = true;
+			break;
+		}
+	} ZEND_HASH_FOREACH_END();
+	if (needs_duplication) {
+		HashTable *duplicated = zend_new_array(zend_hash_num_elements(ctor_args));
+		ZEND_HASH_FOREACH_KEY_VAL(ctor_args, index, arg_name, arg_value) {
+			if (arg_name) {
+				zend_hash_add(duplicated, arg_name, arg_value);
+			} else {
+				zval val;
+				if (!Z_ISREF_P(arg_value) && ARG_SHOULD_BE_SENT_BY_REF(constructor, index+1)) {
+					Z_TRY_ADDREF_P(arg_value);
+					ZVAL_NEW_REF(&val, arg_value);
+				} else {
+					ZVAL_COPY_VALUE(&val, arg_value);
+				}
+				zend_hash_index_add(duplicated, index, &val);
+			}
+		} ZEND_HASH_FOREACH_END();
+		return duplicated;
+	} else {
+		GC_TRY_ADDREF(ctor_args);
+		return ctor_args;
+	}
+}
+
+/* Return value MUST be an initialized object */
+static bool pdo_call_fetch_object_constructor(zend_function *constructor, HashTable *ctor_args, zval *return_value)
+{
+	zval retval_constructor_call;
+	zend_fcall_info fci = {
+		.size = sizeof(zend_fcall_info),
+		.function_name = {},
+		.object = Z_OBJ_P(return_value),
+		.retval = &retval_constructor_call,
+		.param_count = 0,
+		.params = NULL,
+		.named_params = ctor_args,
+	};
+	zend_fcall_info_cache fcc = {
+		.function_handler = constructor,
+		.object = Z_OBJ_P(return_value),
+		.called_scope = NULL,
```

Comment:
Why do you set the called scope to NULL?

Reply:
I was trying to understand if called_scope was required, which it seems it is not as it is "guessed" from the object if given. So something to possibly investigate to make a requirement, but yes the called scope should be provided here.